### PR TITLE
bump Zephyr to tip of main

### PR DESCRIPTION
**Pull request by mniestroj**
Update to the newest version of Zephyr. This requires handling of DORMANT network interface states,
which is handled in this PR as well.

### CODE REVIEW

File: samples/common/net_connect.c [hunk 1]
```diff
@@ -73,8 +73,11 @@ void net_connect(void)
 {
 	struct net_if *iface = net_if_get_default();
 
-	LOG_INF("Waiting for interface to be up");
-	wait_for_iface_up(iface);
+	if (IS_ENABLED(CONFIG_GOLIOTH_SAMPLE_WIFI) &&
```
Comment (sam-golioth) [inline samples/common/net_connect.c:76]: We can combine this with the block below at line 82 to make it a little cleaner.

**Review by sam-golioth — APPROVED** [review]

File: samples/common/net_connect.c (hunk 1 above)
Comment (mniestroj) [reply samples/common/net_connect.c]: Actually implemented check was broken. It was supposed to skip waiting for IFACE UP event for "WiFi && Dormant" interfaces, but still wait for IFACE UP for all others. Fixed that now. After the fix your comment about combining with block at 82 line does not apply anymore. But thanks for commenting about it, it allowed me to notice this bug. Tested with `mimxrt1060_evkb` and confirmed it works after the fix.

**Comment by mniestroj** [comment]
There are problems with ESP32 platform and DFU, hence marked this PR as `DNM`. Working on that now.

**Comment by mniestroj** [comment]
Went with another bump of Zephyr. This pulls in `esp32` board name change to `esp32_devkitc_wroom`. All the bumps are in separate commits, as there are separate issues targeted in each. This will also allow to easily bisect between those bumps in case we find regressions because of some of them.

**Comment by mniestroj** [comment]
> Should we consider adding support for other ESP32 boards, now that they're split up? Specifically thinking about the ESP32-DevKitC-WROVER. It can be a separate change.

I would like to keep the amount of supported platforms minimal. We have reduced those in the past, just because it was consuming to much time to make sure every platform worked. Second thing is that I don't even have this HW right now. We can consider more HWs if we make improvements in the HIL in CI and make all the tests autonomous. Right now we are not there yet.

**Pull request merged**